include/stl2/detail/iterator/move_iterator.hpp

```diff
@@ -90,6 +91,33 @@ STL2_OPEN_NAMESPACE {
 				static_cast<void>(++current_)
 			)
 
+			// Experimental support for move_iterator post-increment
+			using __post_inc_t = decltype(current_++);
+			Readable{R}
+			struct __proxy {
+				R __tmp;
+				decltype(auto) operator*()
+				STL2_NOEXCEPT_RETURN(
+					__stl2::iter_move(__tmp)
+				)
+				friend decltype(auto) iter_move(const __proxy& that)
+				STL2_NOEXCEPT_RETURN(
+					__stl2::iter_move(that.__tmp)
+				)
+			};
+
+			STL2_CONSTEXPR_EXT cursor post_increment()
+			noexcept(noexcept(cursor{cursor{__stl2::exchange(current_, __stl2::next(current_))}}))
+			requires Same<__post_inc_t, I>() {
+				return cursor{__stl2::exchange(current_, __stl2::next(current_))};
```

Comment:
> can we prove that `current_` is valid after `next(current_)`, and that the value of `*current_` must be unchanged?

This cursor:

```c++
template<class T>
class cursor {
    std::istream* is_;
    std::shared_ptr<T> ptr_;
public:
    cursor() = default;
    cursor(std::istream* is) : is_{is} {}
    T const& read() const {
        return *ptr_;
    }
    void next() {
        if (!ptr_) ptr_ = std::make_shared<T>();
        if (!(*ptr_ << *is_)) is_ = nullptr;
    }
    cursor post_increment() {
        auto tmp = std::move(*this);
        next();
        return tmp;
    }
};
```
Implements something like `std::istream_iterator<T>` that uses remote state to minimize iterator copy cost for large `T`. `operator++(int)` conforms to the `Readable` requirement by saving state that is corrupted by calling `operator++()` on any copy of the cursor. I'm reasonably certain that (a) `basic_iterator<cursor<T>>` is a conforming `input_iterator`, and (b) it breaks this `move_iterator`'s `operator++(int)`.